test/controllers/books_controller_test.rb

```diff
@@ -55,6 +73,17 @@ class BooksControllerTest < ActionDispatch::IntegrationTest
     assert_equal 'reserved', book.reload.status
   end
 
+  test 'PATCH/PUT/users/:user_id/books/:id should not edit book' do
+    user = users(:hoffman)
+    book = books(:becoming)
+
+    sign_in user
+
+    patch user_book_url(book.user, book), params: { book: { status: 'reserved' } }
```

Comment:
Not a big deal, but you can also use the `assert_no_changes` helper here. E.g.
```ruby
assert_no_changes, 'book.status' do
  patch ...
end
```

Reply:
Good idea! I changed it.